clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/view/EcsInstanceProvider.java

```diff
@@ -0,0 +1,171 @@
+
+  @Autowired
+  private AccountCredentialsProvider accountCredentialsProvider;
+
+  @Autowired
+  private AmazonClientProvider amazonClientProvider;
+
+
+  @Override
+  public String getCloudProvider() {
+    return cloudProvider;
+  }
+
+  @Override
+  public EcsInstance getInstance(String account, String region, String id) {
+    if (!isValidId(id, region))
+      return null;
+
+    EcsInstance ecsInstance = null;
+
+    AWSCredentialsProvider awsCredentialsProvider = getCredentials(account).getCredentialsProvider();
+    AmazonECS amazonECS = amazonClientProvider.getAmazonECS(account, awsCredentialsProvider, region);
+    AmazonEC2 amazonEC2 = amazonClientProvider.getAmazonEC2(account, awsCredentialsProvider, region);
+
+    Task ecsTask = getTask(amazonECS, id);
+    InstanceStatus instanceStatus = getEC2InstanceStatus(amazonEC2, getContainerInstance(amazonECS, ecsTask));
+
+    if (ecsTask != null && instanceStatus != null) {
+      ecsInstance = new EcsInstance(id, ecsTask, instanceStatus);
+    }
+
+    return ecsInstance;
+  }
+
+  @Override
+  public String getConsoleOutput(String account, String region, String id) {
+    return null;
+  }
+
+  private List<String> getAllClusters(AmazonECS amazonECS) {
+    ListClustersResult listClustersResult = amazonECS.listClusters();
+    List<String> clusterList = listClustersResult.getClusterArns();
+    while (listClustersResult.getNextToken() != null) {
+      listClustersResult = amazonECS.listClusters(
+        new ListClustersRequest().withNextToken(listClustersResult.getNextToken())
+      );
+      clusterList.addAll(listClustersResult.getClusterArns());
+    }
+    return clusterList;
+  }
+
+  private NetflixAmazonCredentials getCredentials(String account) {
+    NetflixAmazonCredentials accountCredentials = null;
+    for (AccountCredentials credentials: accountCredentialsProvider.getAll()) {
+      if (credentials.getName().equals(account)) {
+        if (credentials instanceof NetflixAmazonCredentials) {
+          accountCredentials = (NetflixAmazonCredentials) credentials;
+          break;
+        }
+      }
+    }
+
+    if (accountCredentials == null) {
+      throw new NotFoundException(
+        String.format("AWS account %s was not found.  Please specify a valid account name", account)
+      );
+    }
+
+    return accountCredentials;
+  }
+
+  private boolean isValidId(String id, String region) {
+    String id_regex = "[\\da-f]{8}-[\\da-f]{4}-[\\da-f]{4}-[\\da-f]{4}-[\\da-f]{12}";
+    String id_only = String.format("^%s$", id_regex);
+    String arn = String.format("arn:aws:ecs:%s:\\d*:task/%s", region, id_regex);
+    return id.matches(id_only) || id.matches(arn);
+  }
+
+  private Task getTask(AmazonECS amazonECS, String taskId) {
+    Task task = null;
+
+    List<String> queryList = new ArrayList<>();
+    queryList.add(taskId);
+
+    for (String cluster: getAllClusters(amazonECS)) {
+      List<Task> taskList = amazonECS.describeTasks(
```

Comment:
As a stylistic comment, if you extract `new DescribeTasksRequest().withCluster(cluster).withTasks(queryList))` to a variable using option+command+v, then you have easier to debug code, as you can inspect that particular line

It also makes the code a bit easier to follow, albeit more verbose.  It is a style comment though, you can easily find people saying the opposite of what I just wrote :)

Reply:
Updated